File: camerax_android/lib/src/legacy/core/aspect_ratio.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension AspectRatioArgs on $interface.AspectRatio {
  $native.AspectRatio get obj {
    switch (this) {
      case $interface.AspectRatio.ratioDefault:
        return $native.AspectRatio.ratioDefault;
      case $interface.AspectRatio.ratio4_3:
        return $native.AspectRatio.ratio4_3;
      case $interface.AspectRatio.ratio16_9:
        return $native.AspectRatio.ratio16_9;
    }
  }
}

extension AspectRatioObj on $native.AspectRatio {
  $interface.AspectRatio get args {
    switch (this) {
      case $native.AspectRatio.ratioDefault:
        return $interface.AspectRatio.ratioDefault;
      case $native.AspectRatio.ratio4_3:
        return $interface.AspectRatio.ratio4_3;
      case $native.AspectRatio.ratio16_9:
        return $interface.AspectRatio.ratio16_9;
    }
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/core/AspectRatio.kt
package dev.hebei.camerax_android.legacy.core

import dev.hebei.camerax_android.legacy.AspectRatio

val AspectRatio.obj
    get() = when (this) {
        AspectRatio.RATIO_DEFAULT -> androidx.camera.core.AspectRatio.RATIO_DEFAULT
        AspectRatio.RATIO4_3 -> androidx.camera.core.AspectRatio.RATIO_4_3
        AspectRatio.RATIO16_9 -> androidx.camera.core.AspectRatio.RATIO_16_9
    }

val Int.aspectRatioArgs
    get() = when (this) {
        androidx.camera.core.AspectRatio.RATIO_DEFAULT -> AspectRatio.RATIO_DEFAULT
        androidx.camera.core.AspectRatio.RATIO_4_3 -> AspectRatio.RATIO4_3
        androidx.camera.core.AspectRatio.RATIO_16_9 -> AspectRatio.RATIO16_9
        else -> throw IllegalArgumentException()
    }